ERCS/erc-7846.md

```diff
@@ -0,0 +1,168 @@
+
+## Specification
+
+### `wallet_connect`
+
+Request the user to connect a single account and optionally confirm chain support and add capabilities.
+
+#### RPC Specification
+
+For each chain requested by the app, wallets MUST return a mapped capabilities object if the chain is supported and SHOULD return an empty object if no capabilities exist. Wallets MUST NOT return mapped capabilities objects for chains they do not support. If an app does not declare chains they would like to confirm support for, the wallet can return any chains it wishes to declare support for.
+
+```typescript
+type WalletConnectParams = [{
+  version: string;
+  capabilities?: Record<string,any>; // optional connection capabilities
+}]
+
+type WalletConnectResult = {
+  accounts: {
+    address: `0x${string}`; // connected account address
+    capabilities: Record<string,any>; // results of this connection request's connection capabilities
+  }[]
+}
+```
+
+#### Example Parameters
+
+```json
+[{
+  "version": "1",
+  "capabilities": {
+    "signInWithEthereum": {
+      "nonce": "12345678",
+      "chainId": "0x1"
+    }
+  }
+}]
+```
+
+#### Example Result
+
+```json
+{
+  "accounts": [
+    {
+      "address": "0x...",
+      "capabilities": {
+        "signInWithEthereum": {
+          "message": "app.com wants you to sign in with your Ethereum account:\n0x...",
+          "signature": "0x..."
+        }
+      }
+    }
+  ]
+}
+```
+
+### `signInWithEthereum` Capability
+
+Adds authentication using the [ERC-4361](https://eips.ethereum.org/EIPS/eip-4361) Sign In WIth Ethereum standard.
+
+#### Capability Specification
+
+Same as ERC-4361 specification with minor modifications: 
+* The casing of multi-word fields has been adjusted to mixedCase instead of hyphen-case. Resources are an array field. 
+* The account address returned by `wallet_connect` MUST be the same address that is auto-inserted into the SIWE message.
+* `version` is optional and defaults to an accepted version defined in ERC-4361 if not provided.
+* `domain` is optional and defaults to the domain of the requesting app if not provided.
+* `uri` is optional and defaults to the uri of the requesting app if not provided.
+* `issuedAt` is optional and defaults to the current time if not provided.
+
+The wallet MUST return a properly formatted ERC-4361 message that exactly matches the requested parameters and a signature over the EIP-191 hash of the message. The app SHOULD also verify that the two match for security.
+
+```typescript
+type SignInWithEthereumCapabilityParams = {
+  nonce: string,
+  chainId: string, // EIP-155 hex-encoded
+  version?: string,
+  scheme?: string,
+  domain?: string,
+  uri?: string,
+  statement?: string
+  issuedAt?: string
+  expirationTime?: string,
+  notBefore?: string,
+  requestId?: string,
+  resources?: string[]
+}
+
+type SignInWithEthereumCapabilityResult = {
+  message: string, // formatted SIWE message
```

Comment:
Might also be nice to add a `payload` property so consumers don't have to parse the `message` string to figure out the inferred parameters (e.g. `address`, `domain`, etc)?
```suggestion
  message: string, // formatted SIWE message
  payload: {
    address: `0x{string}`,
    nonce: string,
    ...
  },
```